x/core/keeper/gas_meter.go

```diff
@@ -63,9 +62,82 @@ func (k Keeper) DistributionsFromGasMeter(ctx sdk.Context, reqID string, reqHeig
 	return dists
 }
 
+func (k Keeper) ProcessDistributions(ctx sdk.Context, dists []types.Distribution, dr *types.DataRequest, minimumStake math.Int) error {
+	// Distribute in order.
+	denom, err := k.stakingKeeper.BondDenom(ctx)
+	if err != nil {
+		return err
+	}
+
+	remainingEscrow := dr.Escrow
+
+	// TODO Events
+	var amount math.Int
+	for _, dist := range dists {
+		switch {
+		case dist.Burn != nil:
+			amount = math.MinInt(dist.Burn.Amount, remainingEscrow)
+			err = k.bankKeeper.BurnCoins(ctx, types.ModuleName, sdk.NewCoins(sdk.NewCoin(denom, amount)))
+			if err != nil {
+				return err
+			}
+
+		case dist.DataProxyReward != nil:
+			amount = math.MinInt(dist.DataProxyReward.Amount, remainingEscrow)
+			payoutAddr, err := sdk.AccAddressFromBech32(dist.DataProxyReward.PayoutAddress)
+			if err != nil {
+				// Should not be reachable because the address has been validated.
+				return err
+			}
+			err = k.bankKeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, payoutAddr, sdk.NewCoins(sdk.NewCoin(denom, amount)))
+			if err != nil {
+				return err
+			}
+
+		case dist.ExecutorReward != nil:
+			amount = math.MinInt(dist.ExecutorReward.Amount, remainingEscrow)
+			staker, err := k.GetStaker(ctx, dist.ExecutorReward.Identity)
+			if err != nil {
+				return err
```

Comment:
The contract did not error in this case. IT would simply emit an error event and move on to the next iteration of the loop. The return statement would instead exit the function early not doing the payouts at all.

Or is this another unreachable situation due to earlier checks? Or can executors ever unregister so their proxy no longer exists?

TLDR this is a change from the contract.

Reply:
Good point, will address this in a new PR